target/mips/translate.c

```diff
@@ -6554,31 +6559,17 @@ static void gen_compute_branch(DisasContext *ctx, uint32_t opc,
 #ifndef TARGET_CHERI
     case OPC_JALX:
 #endif
+    {
         /* Jump to immediate */
-
 #ifdef TARGET_CHERI
-        /*
-         * The branch target address is not known since the PCC.base can
-         * change.  Therefore treat this like a JR/JALR and compute the
-         * branch target from the current PCC.base.
-         *
-         * ((ctx->base.pc_next + insn_bytes) - PCC.base) >> 28 << 28 | offset +
-         * PCC.base
-         */
-        tcg_gen_movi_i64(t0, ctx->base.pc_next + insn_bytes);
-        tcg_gen_ld_i64(t1, cpu_env, offsetof(CPUMIPSState, active_tc.PCC) +
-            offsetof(cap_register_t, cr_base));
-        tcg_gen_sub_i64(t0, t0, t1);
-        tcg_gen_shri_i64(t0, t0, 28);
-        tcg_gen_shli_i64(t0, t0, 28);
-        tcg_gen_ori_i64(t0, t0, offset);
-        tcg_gen_add_i64(btarget, t0, t1);
-        GEN_CCHECK_BTARGET(cpu_env);
+        btgt = (((ctx->base.pc_next + insn_bytes - ctx->pcc_base) >> 28) << 28) |
```

Comment:
Can we just do
```
#ifdef TARGET_CHERI
target_ulong pcc_base = ctx->pcc_base;
#else
target_ulong pcc_base = 0;
#endif
```
and then use `pcc_base` unconditionally? (and preferably by adapting the upstream mips variant with the mask rather than the reversed-from-tcg bit shifting. Also preferably with parentheses to disambiguate the precedence of | and +.

Reply:
Sounds good. Will fix.